internal/impl/aws/input_kinesis.go

```diff
@@ -692,6 +648,8 @@ func (k *kinesisReader) runExplicitShards() {
 	}
 }
 
+// This function attempts to try and claim shards only if the number of shards already claimed is less than the max number of shards
+// It returns the number of shards it has successfully claimed and any errors it encountered
```

Comment:
nit: I know not part of this PR but I think when an error happen in this function here:
```
sequence, err := k.checkpointer.Claim(k.ctx, streamID, shardID, clientID)
			if err != nil {
				if k.ctx.Err() != nil {
					return 0, err
				}
```
We should still return the number of so far claimed shards rather than 0

Reply:
oh true, its fixed now :)